### PR TITLE
[MAINT] C style

### PR DESCRIPTION
**Pull request by rafaeldtinoco**
## Initial Checklist

- [ ] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

Introduce clang-format and makefile targets to manage code formatting (C and Go) and static checking (go).

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

No logical changes were made.

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [ ] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented all functions/methods created explaining what they do.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

**Comment by rafaeldtinoco** [comment]
@yanivagman this was already discussed between you an I. There are multiple times reviews take longer because of style changes being needed. This will enforce style AND it is very easy to execute `astyle` before submitting a PR, guaranteeing our code style is kept.

I'm keeping something close to kernel source style, and most of the upstream C projects, but I'm using "4 spaces" instead of no expanded tabs, to allow our code to be more compact (and more code per line, being enforced at 100 cols).

**Comment by rafaeldtinoco** [comment]
The diff will be shown if a style change is needed:

image

**Review by danielpacak — DISMISSED** [review]
Overall looks great! I think the PR needs rebasing. Also see my comments about Make target or script to run checks locally, independently of GitHub Actions runner.

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -51,6 +51,28 @@ jobs:
         with:
           go-version: ${{ env.GO_VERSION }}
           opa-version: ${{ env.OPA_VERSION }}
+      - name: Check C code style
+        run: |
```
[the remark this answers is not in the document]
Comment (rafaeldtinoco) [reply .github/workflows/pr.yaml:55]: Totally agreed and has been requested by @AsafEitani as well. I'll do it.

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -51,6 +51,28 @@ jobs:
...
+          command="astyle --style=kr --indent=spaces=4 --attach-inlines "
```
[the remark this answers is not in the document]
Comment (rafaeldtinoco) [reply .github/workflows/pr.yaml:56]: I can provide that. It was one of my doubts, tbh. I can force single line conditions to always have brackets, for sure.

[the remark this answers is not in the document]
Comment (rafaeldtinoco) [reply .github/workflows/pr.yaml:56]: My problem with that is this: I use K&R as a base and change things on top of that. I can force removing braces from single line condition checkers, but I cannot add braces to oneline single checkers. 

And, 1TBS styling gives me bad things such as:

image

Which aren't used in other projects afaik, and are weird to me.

[the remark this answers is not in the document]
Comment (rafaeldtinoco) [reply .github/workflows/pr.yaml]: I have changed astyle to clang-format, which is better tool, follows C formatting conventions (not enforcing undocumented styles) and better maintained.

[the remark this answers is not in the document]
Comment (rafaeldtinoco) [reply .github/workflows/pr.yaml]: We can revisit formatting changes when we start splitting tracee.bpf.c file.

**Pull request merged**